crypto/crypto.go

```diff
@@ -77,15 +77,15 @@ func Decrypt(base64EncodedCiphertext string, key []byte) []byte {
 }
 
 // DeriveKey derives a 32 byte encryption key from a password and identifier.
-func DeriveKey(password, identifier []byte) []byte {
-	derivedKey, _ := scrypt.Key(password, identifier, 1<<18, 8, 1, 32)
+func DeriveKey(password, identifier []byte, costFactor map[string]int) []byte {
+	derivedKey, _ := scrypt.Key(password, identifier, 1<<uint(costFactor["N"]), costFactor["r"], costFactor["p"], 32)
```

Comment:
Validating user inputs is always a good thing - I doubt the scrypt error is very user friendly, either